src/components/ContextMenu/index.js

```diff
@@ -22,33 +22,36 @@ class ContextMenu {
         if (!args) args = {};
 
         this._menu = new Container({ dom: args.dom });
+        this._menu.contextMenu = this;
+        this.args = args;
         this._menu.class.add(CLASS_ContextMenu);
+        var menu = this._menu;
 
         var removeMenu = () => {
             this._menu.class.remove(CLASS_ContextMenu_active);
             document.removeEventListener('click', removeMenu);
         };
 
-        if (args.dom) {
-            args.dom.parentElement.addEventListener('contextmenu', (e) => {
-                e.preventDefault();
-                this._menu.class.add(CLASS_ContextMenu_active);
-                var maxMenuHeight = args.items.length * 27.0;
-                var maxMenuWidth = 150.0;
-                var left = e.clientX;
-                var top = e.clientY;
-                if (maxMenuHeight + top > window.innerHeight) {
-                    var topDiff = (maxMenuHeight + top) - window.innerHeight;
-                    top -= topDiff;
-                }
-                if (maxMenuWidth + left > window.innerWidth) {
-                    var leftDiff = (maxMenuWidth + left) - window.innerWidth;
-                    left -= leftDiff;
-                }
-                args.dom.setAttribute("style", `left: ${left}px; top: ${top}px`);
-                document.addEventListener('click', removeMenu);
-            });
-        }
+        var triggerElement = args.triggerElement || args.dom.parentElement;
+        triggerElement.addEventListener('contextmenu', (e) => {
```

Comment:
Is it possible for `triggerELement` to be `undefined` here?

Reply:
Oops yeah i'll add the conditional back in!